File: bidding-engine/src/main/java/com/omnibus/bidding/rules/BiddingFacts.java
package com.omnibus.bidding.rules;

import com.omnibus.bidding.domain.HandProfile;

public record BiddingFacts(
    String system,
    String auction,
    String hand,
    int hcp,
    int spades,
    int hearts,
    int diamonds,
    int clubs,
    boolean balanced
) {

    public static BiddingFacts from(HandProfile handProfile, String auction, String system) {
        return new BiddingFacts(
            system,
            auction == null ? "" : auction,
            handProfile.normalizedHand(),
            handProfile.hcp(),
            handProfile.spades(),
            handProfile.hearts(),
            handProfile.diamonds(),
            handProfile.clubs(),
            handProfile.balanced()
        );
    }
}

File: services/core/rule-engine/src/main/kotlin/pl/writeonly/omnibus/rule/rules/BiddingFacts.kt
package pl.writeonly.omnibus.rule.rules

import pl.writeonly.omnibus.rule.domain.HandProfile

data class BiddingFacts(
    val system: String,
    val auction: String,
    val hand: String,
    val hcp: Int,
    val spades: Int,
    val hearts: Int,
    val diamonds: Int,
    val clubs: Int,
    val balanced: Boolean,
) {
    companion object {
        fun from(handProfile: HandProfile, auction: String?, system: String): BiddingFacts =
            BiddingFacts(
                system = system,
                auction = auction ?: "",
                hand = handProfile.normalizedHand,
                hcp = handProfile.hcp,
                spades = handProfile.spades,
                hearts = handProfile.hearts,
                diamonds = handProfile.diamonds,
                clubs = handProfile.clubs,
                balanced = handProfile.balanced,
            )
    }
}

